jquery.serializejson.js

```diff
@@ -194,6 +214,21 @@
       return typeFromDataAttr || null;
     },
 
+    // Find an input in the $form with the same name,
+    // and get the data-skip-empty attribute.
+    // Returns null if none found.
+    // Returns boolean value of first data-skip-empty found if many inputs have the same name.
+    tryToFindSkipFalsyFromDataAttr: function(name, $form) {
+      var escapedName, selector, $input, skipFalsyFromDataAttr;
+      escapedName = name.replace(/(:|\.|\[|\]|\s)/g,'\\$1'); // every non-standard character need to be escaped by \\
+      selector = '[name="' + escapedName + '"]';
+      $input = $form.find(selector).add($form.filter(selector));
+      skipFalsyFromDataAttr = $input.attr('data-skip-falsy'); // NOTE: this returns only the first $input element if multiple are matched with the same name (i.e. an "array[]"). So, arrays with different element types specified through the data-value-type attr is not supported.
+      if (skipFalsyFromDataAttr && (skipFalsyFromDataAttr === "true" || skipFalsyFromDataAttr == "false"))
+        return (skipFalsyFromDataAttr === "true")
+      return null;
+    },
```

Comment:
I would change the logic to: `return skipFalsyFromDataAttr !== "false")`, so anything works as a true value except for the explicit "false" that is meant to override the general option. This is better even if the documentation says to only use "true" or "false", because sometimes people makes typos or they prefer to use "on" instead of "true", in which case it's better to assume a true value than ignoring it (IMO). This will also make it work if the attribute is specified without a value (i.e. `<input data-skip-falsy>` should be assumed enabled).

```js
      if (skipFalsyFromDataAttr != null) {
        return skipFalsyFromDataAttr !== "false";
      }
      return null;
```